### ISSUE TITLE
4.8.0 doesn't allow content outside of tasks

### ISSUE THREAD

**Issue by codykonior** [psake/psake#277]
I had a psakefile.ps1 which was working on 4.7.4 but failed after the 4.8.0 upgrade.

## Expected Behavior
On 4.7.4 this works.

```
$ErrorActionPreference = "Stop"
Set-StrictMode -Version Latest

Task default -Depends Test

Task Test {
    "Hi"
}
```

On 4.8.0:

```
PS C:\temp> invoke-psake
psake version 4.8.0
Copyright (c) 2010-2018 James Kovacs & Contributors

Error: 30/04/2019 10:51:44 AM: 
At C:\temp\psakefile.ps1:4 char:1 + Task default -Depends Test + ~~~~~~~~~~~~~~~~~~~~~~~~~~ [<<==>>] Exception: The property 'default' cannot be found on this object. Verify that the property exists.
```

## Context
I set those error conditions at the top of the script because it then sets all of the task scopes. This can't be done in TaskSetup because it doesn't if done there it doesn't extend beyond that scope.

I've worked around it by removing the error conditions from the top of the script and inserting them into every single task block but it's a lot of duplicate code I'd like to go back to normal again.

## Your Environment
Windows PowerShell 5.1

**Comment by devblackops** [comment]
Thanks for submitting the issue @codykonior. I'll dig into this.

**Comment by devblackops** [comment]
@codykonior This has been fixed in 348ba3846d3bdfff520771048449afd202e7dd74. The issue was if you set strict mode in your psake file, psake itself must also be written to pass strict mode, and I was referencing some hashtable keys in a way that will throw an error in strict mode. This has been fixed and I've also put in a new test file that should hopefully validate strict mode works going forward.

Thanks for catching this!
